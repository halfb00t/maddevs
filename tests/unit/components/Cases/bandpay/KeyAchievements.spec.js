import { render, screen } from '@testing-library/vue'
import { createLocalVue, mount } from '@vue/test-utils'
import KeyAchievements from '@/components/Cases/bandpay/KeyAchievements'
import '../../../__mocks__/intersectionObserverMock'
// eslint-disable-next-line import/order
import lazyLoad from 'nuxt-lazy-load/lib/module'

const mocks = {
  $getMediaFromS3: () => 'img.jpg',
}

const directives = {
  'lazy-load': () => {},
}

const localVue = createLocalVue()
localVue.directive('lazy-load', lazyLoad)
jest.mock('nuxt-lazy-load/lib/module')

describe('BandPay KeyAchievements component', () => {
  it('should render correctly', () => {
    const { container } = render(KeyAchievements, {
      mocks,
      directives,
      localVue,
    })

    expect(container).toMatchSnapshot()
  })

  it('should render with text', () => {
    render(KeyAchievements, {
      mocks,
      directives,
      localVue,
    })

    expect(screen.getByText(/Key achievements/i).className).toBeTruthy()
    expect(screen.getByText(/Mad Devs team is a fully-fledged platform where/i).className).toBeTruthy()
    expect(screen.getByText(/While working on the BandPay solution, we managed/i).className).toBeTruthy()
    expect(screen.getByText(/Easy onboarding/i).className).toBeTruthy()
    expect(screen.getByText(/we prepared sufficient project documentation/i).className).toBeTruthy()
    expect(screen.getByText(/Consequently, it takes around 2 days to onboard/i).className).toBeTruthy()
    expect(screen.getByText(/Increased project visibility/i).className).toBeTruthy()
    expect(screen.getByText(/Mad Devs works in close collaboration with our partners from Peklo/i).className).toBeTruthy()
    expect(screen.getByText(/We added a news feed, which shows all recent activities/i).className).toBeTruthy()
    expect(screen.getByText(/As a result, BandPay is 100% visible and transparent/i).className).toBeTruthy()
  })

  it('should render youtube video', () => {
    const wrapper = mount(KeyAchievements, {
      mocks,
      directives,
      localVue,
    })

    const videoFrame = wrapper.find('.case_iframe')
    expect(videoFrame.element.getAttribute('data-src'))
      .toBe('https://www.youtube.com/embed/a0V5qU5oUfA')
  })

  describe('KeyAchievements images', () => {
    it('should render images', () => {
      render(KeyAchievements, {
        mocks,
        directives,
        localVue,
      })

      screen.getAllByTestId('test-picture-img').forEach(item => {
        expect(item.getAttribute('data-src')).toBe(mocks.$getMediaFromS3())
      })
    })
  })
})
