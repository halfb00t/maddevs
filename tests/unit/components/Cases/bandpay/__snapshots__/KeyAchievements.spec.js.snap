// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BandPay KeyAchievements component should render correctly 1`] = `
<div>
  <section
    class="m-96_bottom media-m-48_bottom"
  >
    <section
      class="container_regular m-48_bottom media-m-24_bottom"
    >
      <h3
        class="case_title_h2 m-12_bottom"
      >
        
      Key achievements
    
      </h3>
       
      <p
        class="case_paragraph"
      >
        
      The outcome of the work completed by the Mad Devs team is a fully-fledged platform where users can build
      partnerships, make agreements, pay, and get paid. BandPay safeguards its users and ensures that everyone
      remains satisfied with the final results, such as the desired deliverables and timely payments.
    
      </p>
    </section>
     
    <div
      class="case_iframe-wrapper m-auto m-48_bottom media-m-24_bottom"
      data-aos="fade-up"
      data-aos-anchor-placement="bottom"
    >
      <iframe
        allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture"
        allowfullscreen="allowfullscreen"
        class="case_iframe"
        data-src="https://www.youtube.com/embed/a0V5qU5oUfA"
        title="YouTube video player"
      />
    </div>
     
    <section
      class="container_regular m-48_bottom media-m-24_bottom"
    >
      <p
        class="case_paragraph m-96_bottom media-m-48_bottom"
      >
        
      While working on the BandPay solution, we managed to simplify onboarding and improve project visibility.
    
      </p>
       
      <h4
        class="case_title_h4 m-12_bottom"
      >
        
      Easy onboarding
    
      </h4>
       
      <p
        class="case_paragraph m-24_bottom media-m-12_bottom"
      >
        
      To facilitate the onboarding process, we prepared sufficient project documentation and adopted
      a containerisation strategy using Docker container technology. Once assigned, a developer, be it a frontend
      or backend engineer, can quickly test and deploy the project regardless of the operating system.
      In addition, we obtained good test coverage for the code (65% coverage for 82,000 lines of the backend code).
    
      </p>
       
      <p
        class="case_paragraph m-48_bottom media-m-24_bottom"
      >
        
      Consequently, it takes around 2 days to onboard a new team member in the BandPay project,
      while in other projects, the onboarding process may last up to 2 weeks.
    
      </p>
       
      <h4
        class="case_title_h4 m-12_bottom"
      >
        
      Increased project visibility
    
      </h4>
       
      <p
        class="case_paragraph m-24_bottom media-m-12_bottom"
      >
        
      Mad Devs works in close collaboration with our partners from Peklo, a digital marketing studio engaged
      in BandPay’s marketing strategies and campaigns, including metrics and analytics. Our engineers implemented
      a variety of metrics in the iOS and Android applications to improve the visibility of the project.
    
      </p>
       
      <p
        class="case_paragraph"
      >
        
      We added a news feed, which shows all recent activities on the platform, such as successful and unsuccessful
      payouts, generated agreements, registered and banned accounts, as well as other activities and events.
      We also added regular reports that provide information about the key metrics and alerts that notify about errors.
    
      </p>
    </section>
     
    <div
      class="container_middle"
    >
      <div
        class="case_image-wrapper m-48_bottom media-m-24_bottom case_box-shadow"
      >
        <picture>
          <source
            class="image grey-background"
            data-srcset="img.jpg ,img.jpg"
            type="image/webp"
          />
           
          <img
            alt="Project visibility"
            class="image grey-background"
            data-src="img.jpg"
            data-srcset="img.jpg"
            data-testid="test-picture-img"
            height="16.45"
            width="1026"
          />
        </picture>
         
        <div
          class="container_middle__main-image"
        >
          <picture>
            <source
              class="image"
              data-srcset="img.jpg ,img.jpg"
              type="image/webp"
            />
             
            <img
              alt="Project visibility"
              class="image"
              data-src="img.jpg"
              data-srcset="img.jpg"
              data-testid="test-picture-img"
              height="595.81"
              width="1028"
            />
          </picture>
        </div>
      </div>
    </div>
     
    <div
      class="container_regular m-48_bottom media-m-24_bottom"
    >
      <p
        class="case_paragraph"
      >
        
      As a result, BandPay is 100% visible and transparent for both the business owner and developers, who can see
      everything happening in the project. This system allows for tracking the activity of the users and collecting
      their feedback in real time. Moreover, our developers can instantly detect and fix problems, even before users
      report them. For this purpose, we extensively apply Sentry.io—a helpful tool for error handling and alerting.
    
      </p>
    </div>
     
    <div
      class="container_middle"
    >
      <div
        class="case_image-wrapper m-48_bottom media-m-24_bottom case_box-shadow"
      >
        <picture>
          <source
            class="image grey-background"
            data-srcset="img.jpg ,img.jpg"
            type="image/webp"
          />
           
          <img
            alt="Project sentry"
            class="image grey-background"
            data-src="img.jpg"
            data-srcset="img.jpg"
            data-testid="test-picture-img"
            height="16.45"
            width="1026"
          />
        </picture>
         
        <div
          class="container_middle__main-image"
        >
          <picture>
            <source
              class="image"
              data-srcset="img.jpg ,img.jpg"
              type="image/webp"
            />
             
            <img
              alt="Project sentry"
              class="image"
              data-src="img.jpg"
              data-srcset="img.jpg"
              data-testid="test-picture-img"
              height="476.81"
              width="1028"
            />
          </picture>
        </div>
      </div>
    </div>
  </section>
</div>
`;
